cmd/jobsClean.go

```diff
@@ -0,0 +1,132 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-storage-azcopy/common"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	type JobsCleanReq struct {
+		withStatus string
+	}
+
+	commandLineInput := JobsCleanReq{}
+
+	// remove a single job's log and plan file
+	jobsCleanCmd := &cobra.Command{
+		Use:     "clean",
+		Aliases: []string{"cl"},
+		Short:   cleanJobsCmdShortDescription,
+		Long:    cleanJobsCmdLongDescription,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("clean command does not accept arguments")
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			withStatus := common.EJobStatus
+			err := withStatus.Parse(commandLineInput.withStatus)
+			if err != nil {
+				glcm.Error(fmt.Sprintf("Failed to parse --with-status due to error: %s.", err))
+			}
+
+			err = handleCleanJobsCommand(withStatus)
+			if err == nil {
+				if withStatus == common.EJobStatus.All() {
+					glcm.Exit(func(format common.OutputFormat) string {
+						return fmt.Sprintf("Successfully removed all jobs.")
+					}, common.EExitCode.Success())
+				} else {
+					glcm.Exit(func(format common.OutputFormat) string {
+						return fmt.Sprintf("Successfully removed jobs with status: %s.", withStatus)
+					}, common.EExitCode.Success())
+				}
+			} else {
+				glcm.Error(fmt.Sprintf("Failed to remove log/plan files due to error: %s.", err))
+			}
+		},
+	}
+
+	jobsCmd.AddCommand(jobsCleanCmd)
+
+	// NOTE: we have way more job status than we normally need, only show the most common ones
+	jobsCleanCmd.PersistentFlags().StringVar(&commandLineInput.withStatus, "with-status", "All",
+		"only remove the jobs with this status, available values: Cancelled, Completed, Failed, InProgress, All")
```

Comment:
Just a thought: change this to something like include and/or exclude status, and then have the user be able to supply multiple values?

Reply:
Probably more sophistication than we have time for I think.  And, given the limited set of values, users can just run it, say, twice, if they want to delete those with 2 values.

Reply:
Yah same as the keep-days, let's wait for some user feedback before adding more sophistication.